cypress/integration/configuration/productTypes/purchaseWithProductTypes.js

```diff
@@ -0,0 +1,219 @@
+  createCheckout
+} from "../../../apiRequests/Checkout";
+import { getOrder } from "../../../apiRequests/Order";
+import { createTypeProduct } from "../../../apiRequests/Product";
+import { getDefaultChannel } from "../../../utils/channelsUtils";
+import {
+  addPayment,
+  createAndCompleteCheckoutWithoutShipping,
+  createWaitingForCaptureOrder
+} from "../../../utils/ordersUtils";
+import {
+  createProductInChannel,
+  deleteProductsStartsWith
+} from "../../../utils/products/productsUtils";
+import {
+  createShipping,
+  deleteShippingStartsWith
+} from "../../../utils/shippingUtils";
+
+describe("Purchase products with all products types", () => {
+  const startsWith = `CyPurchaseByType`;
+  const name = `${startsWith}${faker.datatype.number()}`;
+  const email = `${startsWith}@example.com`;
+
+  let defaultChannel;
+  let address;
+  let warehouse;
+  let attribute;
+  let category;
+  let shippingMethod;
+  let createProductData;
+
+  before(() => {
+    cy.clearSessionData().loginUserViaRequest();
+    deleteProductsStartsWith(startsWith);
+    deleteShippingStartsWith(startsWith);
+    getDefaultChannel().then(channelResp => (defaultChannel = channelResp));
+    cy.fixture("addresses")
+      .then(addresses => {
+        address = addresses.usAddress;
+        createShipping({
+          channelId: defaultChannel.id,
+          name,
+          address,
+          price: 10
+        });
+      })
+      .then(
+        ({ warehouse: warehouseResp, shippingMethod: shippingMethodResp }) => {
+          warehouse = warehouseResp;
+          shippingMethod = shippingMethodResp;
+        }
+      );
+    createAttribute(name)
+      .then(attributeResp => {
+        attribute = attributeResp;
+        createCategory(name);
+      })
+      .then(categoryResp => {
+        category = categoryResp;
+      });
+  });
+
+  beforeEach(() => {
+    cy.clearSessionData().loginUserViaRequest();
+    createProductData = {
+      channelId: defaultChannel.id,
+      warehouseId: warehouse.id,
+      quantityInWarehouse: 10,
+      attributeId: attribute.id,
+      categoryId: category.id,
+      price: 10
+    };
+  });
+
+  it("should purchase digital product", () => {
+    const digitalName = `${startsWith}${faker.datatype.number()}`;
+    createTypeProduct({
+      name: digitalName,
+      attributeId: attribute.id,
+      shippable: false
+    })
+      .then(productType => {
+        createProductData.name = digitalName;
+        createProductData.productTypeId = productType.id;
+        createProductInChannel(createProductData);
+      })
+      .then(({ variantsList }) => {
+        createAndCompleteCheckoutWithoutShipping({
+          channelSlug: defaultChannel.slug,
+          email,
+          billingAddress: address,
+          variantsList,
+          auth: "token"
+        });
+      })
+      .then(({ order }) => {
+        getOrder(order.id);
+      })
+      .then(order => {
+        expect(order.status).to.be.eq("UNFULFILLED");
```

Comment:
Can you add assertion to check isShippingRequired to all cases? :)

Reply:
I added expect to check isShippingRequired has correct value in order for all cases